Cumulative update

antest1 opened the pull request:
3.0 rev1 update

- tune UI (from feedbacks)
  - improve color contrast for expedition check overlay
  - fix overlay bound issue (#119)
  - tune fairy movement behavior
- optimize https traffic capture feature implementation (based on 2.x codebase)
  - pcapdroid implementation seems not stable for some devices (#117)
- add setting for fairy opacity (transparency)
- fix reported bugs
  -  "Show Fairy only in Kancolle Screen" option did not work for GotoBrowser
  - X button did not work for some devices (#116)
  - LAB popup did not closed when touch X button
- update automation receiver for active sniffer

antest1 commented:
@laplamgor you can work on below tweet separately from this (will merge this branch after your stuff done)

https://x.com/Ship778868288/status/1886979665879031913

laplamgor commented:
[0b51379](https://github.com/antest1/kcanotify/pull/118/commits/0b5137994172fea78b7b3374606401034b91fed8) disabled the fling within the screen, which is not I wanted initially. 

I have another fix if you want to remove the tiny movement animation when single-clicking the fairy.

antest1 commented:
> [0b51379](https://github.com/antest1/kcanotify/pull/118/commits/0b5137994172fea78b7b3374606401034b91fed8) disabled the fling within the screen, which is not I wanted initially.
> 
> I have another fix if you want to remove the tiny movement animation when single-clicking the fairy.

there were many opinions that it was inconvenient to adjust the position since the fairy moves slightly even when simply clicking.

[297eab4](https://github.com/antest1/kcanotify/pull/119/commits/297eab4d47b64fdd4b727a7306d4ef0101255f19) did not solve the issue so maybe need to find other solution to fix the issue.

https://github.com/user-attachments/assets/01a2ec60-a220-438b-aae4-f697ac3ed74d

laplamgor commented:
> > [0b51379](https://github.com/antest1/kcanotify/pull/118/commits/0b5137994172fea78b7b3374606401034b91fed8) disabled the fling within the screen, which is not I wanted initially.
> > I have another fix if you want to remove the tiny movement animation when single-clicking the fairy.
> 
> there were many comments that it was inconvenient to adjust the position since the fairy moves slightly even when simply clicking.
> 
> [297eab4](https://github.com/antest1/kcanotify/pull/119/commits/297eab4d47b64fdd4b727a7306d4ef0101255f19) did not solve the issue so maybe need to find other solution to fix the issue.
> 
>  KakaoTalk_20250206_132356416.mp4

I see... two different problems.

Better to just check `currentTime - lastT` instead.

I will include fix in my PR

laplamgor commented:
new bug: fairy is invisible after new install, until the user enter the setting page for the first time

antest1 commented:
> > > [0b51379](https://github.com/antest1/kcanotify/pull/118/commits/0b5137994172fea78b7b3374606401034b91fed8) disabled the fling within the screen, which is not I wanted initially.
> > > I have another fix if you want to remove the tiny movement animation when single-clicking the fairy.
> > 
> > 
> > there were many comments that it was inconvenient to adjust the position since the fairy moves slightly even when simply clicking.
> > [297eab4](https://github.com/antest1/kcanotify/pull/119/commits/297eab4d47b64fdd4b727a7306d4ef0101255f19) did not solve the issue so maybe need to find other solution to fix the issue.
> > KakaoTalk_20250206_132356416.mp4
> 
> I see... two different problems.
> 
> Better to just check `currentTime - lastT` instead.
> 
> I will include fix in my PR

So after checking, I found that the behavior between actual devices and emulator is different, which may be the reason why I felt more sensitive for fairy movement. Specifically, `MotionEvent.ACTION_MOVE` events occurs much more frequently in the device than the emulator, which makes dx/dy/dt tends to be much more smaller even with small coordination changes.

![vv](https://github.com/user-attachments/assets/7a43ae60-c3e8-4f59-b623-1d933a95c49a)

Therefore, I fixed the implementation to check the difference of values ​​in `lastX` and `lastY` along with dt. (70d3359)
Let me know if you find any problems with current one!


> new bug: fairy is invisible after new install, until the user enter the setting page for the first time

fixed in (3b6d10d).

laplamgor commented:
Device has a higher sampling rate (>60 usually)
And touch pointer is not as steady as a mouse 

A few pixels of threshold makes sense to me 👍
